crates/video/src/vhu_video.rs

```diff
@@ -0,0 +1,391 @@
+                if evt_idx {
+                    // vm-virtio's Queue implementation only checks avail_index
+                    // once, so to properly support EVENT_IDX we need to keep
+                    // calling process_queue() until it stops finding new
+                    // requests on the queue.
+                    loop {
+                        commandq.disable_notification().unwrap();
+                        thread.process_command_queue(commandq)?;
+                        if !commandq.enable_notification().unwrap() {
+                            break;
+                        }
+                    }
+                } else {
+                    // Without EVENT_IDX, a single call is enough.
+                    thread.process_command_queue(commandq)?;
+                }
+            }
+
+            EVENT_Q => {
+                // This queue is used by the device to asynchronously send
+                // event notifications to the driver. Thus, we do not handle
+                // incoming events.
+                warn!("Unexpected event notification received");
+            }
+
+            VIDEO_EVENT => {
+                thread.process_video_event(eventq)?;
+            }
+
+            _ => {
+                warn!("unhandled device_event: {}", device_event);
+                return Err(VuVideoError::HandleUnknownEvent.into());
+            }
+        }
+        Ok(false)
+    }
+
+    fn get_config(&self, _offset: u32, _size: u32) -> Vec<u8> {
+        let offset = _offset as usize;
+        let size = _size as usize;
+
+        let buf = self.config.as_slice();
+
+        if offset + size > buf.len() {
+            return Vec::new();
+        }
+
+        buf[offset..offset + size].to_vec()
+    }
+
+    fn exit_event(&self, _thread_index: usize) -> Option<EventFd> {
+        self.exit_event.try_clone().ok()
+    }
+}
+
+#[cfg(test)]
+pub mod tests {
+    use super::*;
+    use rstest::*;
+    use std::fmt;
+    use std::fs::File;
+    use tempfile::{tempdir, TempDir};
+    use vm_memory::GuestAddress;
+
+    impl fmt::Debug for VuVideoBackend {
+        fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+            f.debug_struct("Point")
+                .field("config", &self.config)
+                .field("exit_event", &self.exit_event)
+                .finish()
+        }
+    }
+
+    pub struct VideoDeviceMock {
+        pub path: String,
+        _dev: File,
+    }
+
+    impl VideoDeviceMock {
+        pub fn new(test_dir: &TempDir) -> Self {
+            let v4l2_device = test_dir.path().join("video.dev").display().to_string();
+            Self {
+                path: v4l2_device.clone(),
+                _dev: File::create(v4l2_device).expect("Could not create a test device file."),
+            }
+        }
+    }
+
+    impl Drop for VideoDeviceMock {
+        fn drop(&mut self) {
+            std::fs::remove_file(self.path.clone()).expect("Failed to clean up test device file.");
```

Comment:
```suggestion
            std::fs::remove_file(&self.path).expect("Failed to clean up test device file.");
```
It takes AsRef<Path> i think